### PR TITLE
Add IJ plugin for reusing Compose in plugins

### PR DESCRIPTION
**Pull request by AlexeyTsvetkov**

### CODE REVIEW

File: tooling/buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+}
+
+repositories {
+    mavenCentral()
+    gradlePluginPortal()
+}
+
+dependencies {
+    compileOnly(gradleApi())
+    implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.10")
```
Comment (olonho) [inline tooling/buildSrc/build.gradle.kts:12]: Maybe use property file for Kotlin version?

File: tooling/buildSrc/src/main/kotlin/idePlugin.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+import org.jetbrains.intellij.tasks.PublishPluginTask
+import org.jetbrains.intellij.tasks.RunPluginVerifierTask
+
+/*
+ * Copyright 2020-2022 JetBrains s.r.o. and respective authors and developers.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+fun Project.intellijPlugin(group: String, fn: IntelliJPluginExtension.() -> Unit = {}) {
+    plugins.apply("org.jetbrains.intellij")
```
Comment (olonho) [inline tooling/buildSrc/src/main/kotlin/idePlugin.kt:14]: What if plugin was already applied?

File: tooling/compose-intellij-platform/gradle.properties [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Opt-out flag for bundling Kotlin standard library.
+# See https://kotlinlang.org/docs/reference/using-gradle.html#dependency-on-the-standard-library for details.
+kotlin.stdlib.default.dependency=false
+
+skiko.version=0.7.0
```
Comment (olonho) [inline tooling/compose-intellij-platform/gradle.properties:5]: Why this version? Is it what we use in 1.0.1?

File: tooling/compose-intellij-platform/sample/base/src/main/kotlin/org/jetbrains/compose/intellij/platform/sample/Theme.kt [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.intellij.ide.ui.LafManager
+import com.intellij.ide.ui.LafManagerListener
+import com.intellij.openapi.application.ApplicationManager
+import java.awt.Color as AWTColor
+import javax.swing.UIManager
+
+val green200 = Color(0xffa5d6a7)
```
Comment (olonho) [inline tooling/compose-intellij-platform/sample/base/src/main/kotlin/org/jetbrains/compose/intellij/platform/sample/Theme.kt:22]: `private val`?

File: tooling/compose-intellij-platform/sample/base/src/main/resources/META-INF/plugin.xml [hunk 5]
```diff
@@ -0,0 +1,5 @@
+<idea-plugin>
+    <id>org.jetbrains.compose.intellij.platform.sample.base</id>
+    <vendor>Sample vendor</vendor>
```
Comment (olonho) [inline tooling/compose-intellij-platform/sample/base/src/main/resources/META-INF/plugin.xml:3]: Guess "JetBrains" would do here.

**Review by olonho — APPROVED** [review]
Overall looks very cool!

File: tooling/compose-intellij-platform/sample/plugin-2/build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+jvmTarget("11")
+
+dependencies {
+    compileOnly(compose.desktop.currentOs)
+}
+
+intellijPlugin(group = "org.jetbrains.compose.intellij.platform.sample.plugin2") {
+    plugins.set(
```
Comment (olonho) [inline tooling/compose-intellij-platform/sample/plugin-2/build.gradle.kts:16]: Would be great to see how out of project configuration would look here? Do we need to publish an additional plugin, right?

File: tooling/buildSrc/src/main/kotlin/idePlugin.kt (hunk 2 above)
Comment (AlexeyTsvetkov) [reply tooling/buildSrc/src/main/kotlin/idePlugin.kt:14]: Nothing will happen

File: tooling/compose-intellij-platform/sample/plugin-2/build.gradle.kts (hunk 6 above)
Comment (AlexeyTsvetkov) [reply tooling/compose-intellij-platform/sample/plugin-2/build.gradle.kts:16]: I don't understand the question

**Pull request merged**